B 10 get order by id in order status new

shaguahill opened the pull request:
1. Created mapper for order with photo list
2. Added method getOrderByIdInStatusNew into Dao and Service
3. Added list with sources to photo
3. Created OrderServlet
4. Added page for order in status new and css for this page
5. Changed Context ro WebContext and added changes to Servlets (Fixed bug with assets)
6. Fixed temporary bug with get Order by id for another statuses (not New)
7. created js file for function "double-click" from all orders page to order by id

Rooman committed:
Merge pull request #51 from Rooman/develop

Release 1

shaguahill committed:
Merge branch 'develop', remote-tracking branch 'origin' into b-10-get-order-by-id-in-order-status-NEW

shaguahill committed:
[b-10] Get order by ID(order status NEW):
1. Created mapper for order with photo list
2. Added method getOrderByIdInStatusNew into Dao and Service
3. Added list with sources to photo
3. Created OrderServlet
4. Added page for order in status new and part of css for this page

shaguahill committed:
[b-10] Get order by ID(order status NEW):
1. Changed Context ro WebContext and added changes to Servlets (Fixed bug with assets)
2. Fixed temporary bug with get Order by id for another statuses (not New)
3. Completed css file for new-order page
4. created js file for function "double-click" from all orders page to order by id

shaguahill committed:
Merge branch 'develop' into b-10-get-order-by-id-in-order-status-NEW

# Conflicts:
#	src/main/webapp/WEB-INF/web.xml

Resolve conflict after merge

shaguahill committed:
[b-10] Get order by ID(order status NEW):
1. Change text in navigation
2. In test-db added one orderPhoto
3. Created test for getOrderByIdInStatusNew

Yuliia-Sh commented on src/main/java/com/photostudio/web/servlet/AccessDeniedServlet.java:
```diff
@@ -19,7 +19,7 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) {
             response.setContentType("text/html;charset=utf-8");
 
-            TemplateEngineFactory.process("access_denied", paramsMap, response.getWriter());
+            TemplateEngineFactory.process(request, response, "access_denied", paramsMap, response.getWriter());
```
Why do you put both response and response.getWriter?

shaguahill replied:
Fixed

Yuliia-Sh commented on src/main/java/com/photostudio/web/servlet/order/OrderServlet.java:
```diff
@@ -0,0 +1,47 @@
+            LOG.info("Request get order page by id in status NEW");
+            String uri = request.getRequestURI();
+            String[] partsOfUri = uri.split("/");
+            int id = Integer.parseInt(partsOfUri[partsOfUri.length - 1]);
```
I thought that orderId should be parameter of request

shaguahill replied:
I use id after "/" because it is single domain abject. We usually use parameters for describe object (like as a filters). It is rules.

shaguahill committed:
Merge branch 'develop' into b-10-get-order-by-id-in-order-status-NEW

# Conflicts:
#	src/main/java/com/photostudio/security/SecurityService.java
#	src/main/java/com/photostudio/web/templater/TemplateEngineFactory.java
#	src/main/webapp/WEB-INF/web.xml

Resolve conflicts

shaguahill committed:
Merge branch 'develop' into b-10-get-order-by-id-in-order-status-NEW

# Conflicts:
#	src/main/java/com/photostudio/dao/OrderDao.java
#	src/main/java/com/photostudio/dao/jdbc/JdbcOrderDao.java
#	src/main/java/com/photostudio/service/OrderService.java
#	src/main/java/com/photostudio/service/impl/DefaultOrderService.java
#	src/main/resources/application.properties

Resolve conflicts

shaguahill committed:
Merge branch 'develop' into b-10-get-order-by-id-in-order-status-NEW

# Conflicts:
#	src/main/java/com/photostudio/dao/OrderDao.java
#	src/main/java/com/photostudio/dao/jdbc/JdbcOrderDao.java
#	src/main/java/com/photostudio/service/OrderService.java
#	src/main/java/com/photostudio/service/impl/DefaultOrderService.java
#	src/main/resources/application.properties

Resolve new conflicts

shaguahill committed:
[b-10] Get order by ID(order status NEW):
1. [b-10-4] Add Servlet for download photo from local storage:
1.1. Created GetPhotoServlet
1.2. Fixed path to photo in another files

shaguahill committed:
[b-10] Get order by ID(order status NEW):
Delete link from button "New order"

shaguahill committed:
[b-10] Get order by ID(order status NEW) (Review Fix):
Remove Writer parameter from process method and fixes another Servlets

ValentynaBondarenko commented on src/main/java/com/photostudio/web/servlet/GetPhotoServlet.java:
```diff
@@ -0,0 +1,36 @@
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) {
+        String prefixPath = request.getContextPath() + PATH_TO_PHOTO;
+
```
add log in this servlet

shaguahill replied:
Fixed

shaguahill committed:
[b-10] Get order by ID(order status NEW) (Review Fix):
Added Logs to GetPhotoServlet

shaguahill committed:
Merge branch 'develop' into b-10-get-order-by-id-in-order-status-NEW

# Conflicts:
#	src/main/java/com/photostudio/dao/OrderDao.java
#	src/main/java/com/photostudio/service/OrderService.java
#	src/main/java/com/photostudio/service/impl/DefaultOrderService.java
#	src/main/java/com/photostudio/web/servlet/order/AllOrdersServlet.java

Resolve conflicts

shaguahill committed:
Merge branch 'develop' into b-10-get-order-by-id-in-order-status-NEW

# Conflicts:
#	src/main/java/com/photostudio/dao/OrderDao.java
#	src/main/java/com/photostudio/service/OrderService.java
#	src/main/java/com/photostudio/service/impl/DefaultOrderService.java
#	src/main/java/com/photostudio/web/servlet/order/AllOrdersServlet.java

Resolve new conflicts

shaguahill committed:
b-10-get-order-by-id-in-order-status-NEW (Fixes after  review):
1. Changed types of LOG in catch parts to error
2. Added to LOGs orderId and pathToPhoto
